tools/fuzzer/platform/imx8-host.c

```diff
@@ -266,7 +266,7 @@ static int imx8_irq_handler(int irq, void *context)
 
 	/* Interrupt arrived, check src */
 	status = imx_mu_read(fuzzer, IMX8_MU13_A_BAR, IMX_MU_xSR);
-
+	status = status << 32 >> 32;
```

Comment:
...why? That just feels extremely wrong. You're essentially clearing the top 32 bits this way.

Reply:
Yeah, because we need to compare the lower 32 bits with a 32 bit number `status & IMX_MU_xSR_GIPn(0)`. The masking you suggested works as well. Changed.

Reply:
Did you see any behavior chance by adding this masking? IMX_MU_xSR_GIPn(0) is a single bit and is unsigned, FWIW. So that masking isn't exactly required for the comparison.

Please check the IMX_MU_xSR_GIPn macro to verify that it is indeed unsigned. If it uses the BIT() macro internally, it should be safe. Otherwise this change needs to be dropped.

Reply:
It does use the `BIT` macro internally - https://github.com/thesofproject/sof/pull/3239/files#diff-9bf68642110c1c3a64a989e8a82a0727R24

Reply:
Alright. Then the masking out of the upper bits isn't necessary anyway. It's not like you're changing data types to gain a modest benefit from _that_. So I don't see the value of this commit. It only introduces a small figment of complexity but is, at least to me, completely unnecessary.

Reply:
Removed it. However, clearing the bits made the code simpler(?) because the MU has 32 bit registers and reading 64 bits from the status register at 0x20 read from 0x20-0x27 while we were only interested in 0x20-0x23. 0x24-0x27 is actually another register - the control register's value. After clearing the status's value would **actually** would've been the status regsiter's value.

Reply:
Yeah, I guess we simply don't have a 32-bit read accessor so reading 64 bits then clearing the top 32 is overcomplicating when you will automatically clear those bits anyway when ANDing with IMX_MU_xSR_GIPn.